src/parsers/manifest/metaplaylist/metaplaylist_parser.ts

```diff
@@ -293,30 +292,24 @@ function createManifest(
     periods.push(...manifestPeriods);
   }
 
-  let duration : number|undefined;
-  if (!isLive) {
-    if (lastEnd === null || firstStart === null) {
-      throw new Error("MPL Parser: can't define duration of manifest.");
-    }
-    duration = lastEnd - firstStart;
-  }
-
   const time = performance.now();
-  const manifest = {
-    availabilityStartTime: 0,
-    clockOffset,
-    suggestedPresentationDelay: 10,
-    duration,
-    id: "gen-metaplaylist-man-" + generateManifestID(),
-    periods,
-    transportType: "metaplaylist",
-    isLive,
-    uris: url == null ? [] :
-                        [url],
-    maximumTime: { isContinuous: false, value: maximumTime, time },
-    minimumTime: { isContinuous: false, value: minimumTime, time },
-    lifetime: mplData.pollInterval,
-  };
+  const manifest = { availabilityStartTime: 0,
```

Comment:
So we do not need `duration` anymore here ?

Reply:
Nope